R/dwdData.R

```diff
@@ -0,0 +1,204 @@
+################################################################################
+#' Download, unzip, untar data from an url
+#'
+#' This function download and untar and/or unzip dataset using url link from available dataset. Prior to download, the function
+#' can check if the files are already locally by using the checksum value available in dbHash.sqlite database.
+#'
+#' @param urltble A data.table that stores available dataset name, url, password and filenames found within each dataset.
+#'        urltble is provided within the package as urls object.
+#' @param datasetName A character string. Represent the dataset of interest for download. datasetName allow to derived
+#'        url and password from the urltble.
+#' @param dfile A vector, given as a character string. Represent filename of interest to download. Default value is
+#'        to download everything using the argument "all".
+#' @param outdir A character string. Indicate output directory where the files will be stored. Default will use the
+#'               modulePath from the sim object, if supplied with a module name, or a temporary location based on the
+#'               url of the file(s).
+#' @param sim A simList simulation object, generally produced by simInit.
+#' @param module A character string representing the names of the module to be loaded for the simulation.
+#' @param checkhash A logical argument. If TRUE, check if file is found locally and cross-check its checksum value
+#'        with value logged in dbHash from previous download. When checksums match, a message raises indicating the file
+#'        is properly downloaded. No download occurs. When checksums don't match or file doesn't exist locally
+#'        (first download), download occurs and checksum compiles. If FALSE, download occurs even if the file exists
+#'        locally. Default is FALSE.
+#' @param dbHash A character string. The path to the database file where checksum values are logged. If the named
```

Comment:
what is the default location for this sqlite db? is the same db always used? if so, how can the user clear out old/stale entries? if a separate db is used (say, in the specified download location per file) then I worry about cluttering up the file system.